Store Playback Information in JSON File.

rynti opened the pull request:
Related Issue: #156

### Description
This feature implements an interface for other applications to read
information about the current song and playback status of the player.

The interface consists of a JSON-formatted file, located at:
%APPDATA%\Google Play Music Desktop Player\playback-information.json

The file includes the following attributes:
- **isActive**: (bool) Whether or not the player is running
- **isPlaying**: (bool) Whether or not a song is currently playing
- **song**: (string) The name of the current song
- **artist**: (string) The artist of the current song
- **album**: (string) The album of the current song
- **imageURL**: (string) An URL to the current song or album image

### Changes

All changes should be self-explanatory, but just to make sure:
- PlaybackAPI class is introduced to manage the current playback status.
- New JavaScript interface setPlaybackStatus is introduced to
  update the "isPlaying" attribute of the playback information.
- The existing JavaScript interface songChangeEvent is used to update
  the song, artist, album, and imageURL attributes of the playback
  information.
- The FormClosing event of the main application is used to set the
  isActive attribute to "false" just before the player closes so the
  file reflects that state properly.

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.svg" height="40" alt="Review on Reviewable"/>](https://reviewable.io/reviews/marshallofsound/google-play-music-desktop-player-unofficial-/157)
<!-- Reviewable:end -->

MarshallOfSound commented:
Also, can we add some information about this feature to the README :smile: 

---

Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[Google Play Music/PlaybackAPI.cs, line 13 \[r1\]](https://reviewable.io:443/reviews/marshallofsound/google-play-music-desktop-player-unofficial-/157#-K90Wa69SpEdy0D27wQ-:-K90Wa69SpEdy0D27wQ0:257112144) ([raw file](https://github.com/marshallofsound/google-play-music-desktop-player-unofficial-/blob/e2af0b217f11f6954b53f42ddb5fbce5e9de8320/Google Play Music/PlaybackAPI.cs#L13)):*
Somewhere else (I believe in our CEF settings) I target the `%APPDATA%\\GPMDP` directory.  Can we be consistent and use that directory?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/marshallofsound/google-play-music-desktop-player-unofficial-/157#-:-K90WpwgJbHPs-W3dyos:-1274167732)*
<!-- Sent from Reviewable.io -->

rynti commented:
In my latest commit 2a42d5d I've added a documentation for the interface, mentioned it in the README.md and changed the path from "Google Play Music Desktop Player" inside %APPDATA% to just "GPMDP" inside %APPDATA%.

MarshallOfSound commented:
Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/marshallofsound/google-play-music-desktop-player-unofficial-/157)*
<!-- Sent from Reviewable.io -->

MarshallOfSound commented:
Review status: all files reviewed at latest revision, all discussions resolved.

---

*[Google Play Music/PlaybackAPI.cs, line 13 \[r1\]](https://reviewable.io:443/reviews/marshallofsound/google-play-music-desktop-player-unofficial-/157#-K90Wa69SpEdy0D27wQ-:-K90wFJgHL10w3oFQI3B:-485063445) ([raw file](https://github.com/marshallofsound/google-play-music-desktop-player-unofficial-/blob/e2af0b217f11f6954b53f42ddb5fbce5e9de8320/Google Play Music/PlaybackAPI.cs#L13)):*
Sweet

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/marshallofsound/google-play-music-desktop-player-unofficial-/157)*
<!-- Sent from Reviewable.io -->
